### PR TITLE
Istio support matrix

### PR DESCRIPTION
**Pull request by bewebi**
# Description

Update the Istio support matrix and add internal documentation for the Istio integration

## Docs changes
- Update support matrix to include Istio 1.18

## Internal docs changes
- Add README to istio kube2e tests, with link to Google Doc about Edge's Istio integration

# Context

#8676 added/validated Istio 1.18 support but neglected to update the support matrix in docs

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: test/kube2e/istio/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Additional context on Istio and how the Gloo integration with Istio works can be found [here](https://docs.google.com/document/d/1g7wq6yBGR6VioNJTz_eA7GLRbijlDasGR3h-eToz7jA)
```
Comment (sam-heilbron) [inline test/kube2e/istio/README.md:1]: This is a google doc that is only available to people within the Solo org, and is linked in our open source repository. Is it possible to extract the contents of the doc and place them here so that anyone can view it?

File: test/kube2e/istio/README.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+- Gateway-proxy
+- Ext-auth-service
+- Rate-limiter
+
+## How does this work?
+Each component involved in the xDS communication will contain an SDS sidecar and potentially an Envoy sidecar as well (for our gateway-proxy pod, which already contains an envoy container, this Envoy sidecar is not necessary)
+
+**Envoy Sidecar**: Responsible for TLS termination, and outgoing encryption. Receives it’s secret information via SDS, talking to the SDS sidecar. The sidecar bootstrap configuration is defined in a ConfigMap for each component.
+
+**SDS Sidecar**: Watches the gloo-mtls-certs TLS secret in kubernetes, and serves the contents via the SDS API to the Envoy sidecar. The SDS code can be found here. Anytime the secret is modified, the contents will be updated dynamically. To automate the updating of secrets, run the [CertGen Job](https://github.com/solo-io/gloo/blob/main/install/helm/gloo/templates/19-gloo-mtls-certgen-job.yaml)
```
Comment (inFocus7) [inline test/kube2e/istio/README.md:21]: nit: missing link maybe? In the sentence:
> The SDS code can be found here

**Review by inFocus7 — APPROVED** [review]

**Review by sam-heilbron — APPROVED** [review]
